### ISSUE TITLE
Parallel jobs don't run on the same git sha

### ISSUE THREAD

**Issue by ssmiech** [soundcloud/pipeline-generator#5]
Since multi jobs don't pass git sha to the downstream dependencies, the downstream jobs will always execute the latest commit from `origin`

given a job config: 
```
{
    "stages": [
        {
            "name": "build",
            "jobs": [
                [
                        { "static_analysis": {"cmd": "echo ''" } },
                        { "compile": {"cmd": "echo ''" } },
                        { "dex_watchdog": {"cmd": "echo ''" } }
                ]
            ],
            "next-stages": [ "test" ]
        },
        {
            "name": "test",
            "jobs": [
                [
                    { "unit_tests": {"cmd": "echo ''" } },
                    { "acceptance_tests": {"cmd": "echo ''" } }
                ]
            ],
            "next-stages": [ "deploy" ]
        },
        {
            "name": "deploy",
            "jobs": [
                { "deploy_alpha": {"cmd": "echo 'deployingAlpha'"} }
            ]
        }
    ],
    "settings": {
        "jenkins-server": "http://localhost:8080",
        "default-name" : "pipelinetest",
        "git-url": "git@github.com:ssmiech/pipeline-test.git",
        "slave-label": "master"
    }
}
```

if there are any commits in between pipeline stages, the stages after commit will be executed with wrong git commit leading to incorrect build artifacts/results.

**Comment by ssmiech** [comment]
I think the solution would be to pass the git-commit to the job phases + pass it further to the downstream dependencies: 
<img width="822" alt="screen shot 2015-09-27 at 15 28 44" src="https://cloud.githubusercontent.com/assets/2083862/10123026/90125792-652c-11e5-8ab0-d71be7de97f7.png">

**Comment by ssmiech** [comment]
+ don't use `trigger other builds`  to trigger other jobs as this one does not pass the git params. 

Now I think all pipelines are affected in the same way, could it be? 

<img width="1096" alt="screen shot 2015-09-27 at 15 30 18" src="https://cloud.githubusercontent.com/assets/2083862/10123033/c70f8648-652c-11e5-8cbb-2227ec4b5592.png">

**Comment by ssmiech** [comment]
Fixed #6
